src/transactions/TransactionFrame.cpp

```diff
@@ -1059,9 +899,11 @@ bool
 TransactionFrame::commonValidPreSeqNum(
     Application& app, AbstractLedgerTxn& ltx, bool chargeFee,
     uint64_t lowerBoundCloseTimeOffset, uint64_t upperBoundCloseTimeOffset,
-    std::optional<FeePair> sorobanResourceFee)
+    std::optional<FeePair> sorobanResourceFee,
+    TransactionResultPayloadPtr resPayload) const
 {
     ZoneScoped;
+    releaseAssertOrThrow(resPayload);
```

Comment:
Can we just accept a reference instead?

Reply:
No, while this is a private function, it calls `checkSorobanResourceAndSetError`, which is public. I think all public functions should still take a pointer over reference (see comments on `checkValid`).